minitwit/src/main/java/WebApplication.java

```diff
@@ -84,6 +80,42 @@ public static String render(Map<String, Object> model, String templatePath) {
         }
     }
 
+    public static Route add_message = (Request request, Response response)  -> {
+        Map<String, Object> model = new HashMap<>();
+//testing
+        try {
+            var db = new SQLite();
+            var conn = db.getConnection();
+
+            if (request.session().attribute(request.queryParams("user_id")) == null) { // Python: # if 'user_id' not in session: #
+                // httpResponse.sendError(HttpServletResponse.SC_UNAUTHORIZED, "Unauthorised"); // Do we need to use this?
+                halt(401);
+            }
+
+            if (request.requestMethod().equals("POST")) { //I'm incredibly unsure of this one too, checked them out online but failed to come out with stuff.
+                var insert = conn.prepareStatement("insert into message (author_id, text, pub_date, flagged)\n" +
+                        "            values (?, ?, ?, 0)");
+
+                insert.setString(1, request.queryParams("author_id"));
```

Comment:
I think the author id should come from the session (which we cant access right now, I just hardcoded mine to 1), instead of whatever the client tells.

Reply:
updated it now, gonna commit
`insert.setString(1, request.session().attribute("author_id"));`